repopool/repo_pool.go

```diff
@@ -143,6 +143,17 @@ func (rp *RepoPool) Mirror(ctx context.Context, remote string) error {
 	return repo.Mirror(ctx)
 }
 
+// QueueMirrorRun is wrapper around repositories QueueMirrorRun method
+func (rp *RepoPool) QueueMirrorRun(remote string) error {
+	repo, err := rp.Repository(remote)
+	if err != nil {
+		return err
+	}
+
+	repo.QueueMirrorRun()
```

Comment:
I think it would be safer to pass in here the rp.ctx, and use that context in the select when putting the message on the channel. This way that call wouldn't get blocked in case the context is cancelled, and so, the consumer of the channel exists.

Reply:
I am sorry I  dont understand this.
`QueueMirrorRun()` is a non blocking function.
our channel is buffered so the caller is not blocked when sync is running..
https://github.com/utilitywarehouse/git-mirror/pull/57/files#diff-cba3bb69ccfed57cee4ff70646c30246388f0c4d2ec17005c8858d1f85e498d0R144-R147

also that function has select statement which will exit when chan is full
https://github.com/utilitywarehouse/git-mirror/pull/57/files#diff-cba3bb69ccfed57cee4ff70646c30246388f0c4d2ec17005c8858d1f85e498d0R475-R485

So call to `QueueMirrorRun` will never block.

Also the reason that channel is only buffered to `1` is because we only care about just 1 trigger. even if there is two consecutive request to run mirror. we just need to do one mirror to satisfy both requests.

Reply:
>also that function has select statement which will exit when chan is full

Riiight.. I didn't notice this. In this case it won't block indeed.
Nevermind then.